src/kinematics.cpp

```diff
@@ -262,11 +262,39 @@ void Kinematics::update_V() {
 
 void Kinematics::update_S() {
     S.setZero(6 * number_of_links, number_of_dofs);
-    int k = 1;
-    for (auto &axis:joint_axis) {
-        S.block(6 * k, k - 1, 6, 1) = axis;
-        k++;
+
+    vector<string> link_names_local = {"upperarm_left", "lowerarm_left", "hand_left",
+                                       "head",
+                                       "upperarm_right", "lowerarm_right", "hand_right",
+                                       };
+    vector<vector<string>> joint_names_local = {
+            {"shoulder_left_axis0", "shoulder_left_axis1", "shoulder_left_axis2"},{"elbow_left_axis0","elbow_left_axis1"},{"wrist_left_axis0", "wrist_left_axis1", "wrist_left_axis2"},
+            {"head_axis0", "head_axis1", "head_axis2"},
+            {"shoulder_right_axis0", "shoulder_right_axis1", "shoulder_right_axis2"},{"elbow_right_axis0","elbow_right_axis1"},{"wrist_right_axis0", "wrist_right_axis1", "wrist_right_axis2"},
+    };
```

Comment:
this goes to the `robots/upper_body/config` folder and should be loaded to the parameter server from a yaml file from there and then read in the constructor or "init" function